panos_upgrade_assurance/firewall_proxy.py

```diff
@@ -617,6 +617,104 @@ def get_routes(self) -> dict:
+        properties that make a key are also available in the value of a dictionary element.
+
+        ```python showLineNumbers title="Sample output"
+        {
+            'default_Peer-Group1_Peer1': {
+                '@peer': 'Peer1',
+                '@vr': 'default',
+                'peer-group': 'Peer-Group1',
+                'peer-router-id': '169.254.8.2',
+                'remote-as': '64512',
+                'status': 'Established',
+                'status-duration': '3804',
+                'password-set': 'no',
+                'passive': 'no',
+                'multi-hop-ttl': '2',
+                'peer-address': '169.254.8.2:35355',
+                'local-address': '169.254.8.1:179',
+                'reflector-client': 'not-client',
+                'same-confederation': 'no',
+                'aggregate-confed-as': 'yes',
+                'peering-type': 'Unspecified',
+                'connect-retry-interval': '15',
+                'open-delay': '0',
+                'idle-hold': '15',
+                'prefix-limit': '5000',
+                'holdtime': '30',
+                'holdtime-config': '30',
+                'keepalive': '10',
+                'keepalive-config': '10',
+                'msg-update-in': '2',
+                'msg-update-out': '1',
+                'msg-total-in': '385',
+                'msg-total-out': '442',
+                'last-update-age': '3',
+                'last-error': 'None',
+                'status-flap-counts': '2',
+                'established-counts': '1',
+                'ORF-entry-received': '0',
+                'nexthop-self': 'no',
+                'nexthop-thirdparty': 'yes',
+                'nexthop-peer': 'no',
+                'config': {'remove-private-as': 'no'},
+                'peer-capability': {
+                    'list': [
+                        {'capability': 'Multiprotocol Extensions(1)', 'value': 'IPv4 Unicast'},
+                        {'capability': 'Route Refresh(2)', 'value': 'yes'},
+                        {'capability': '4-Byte AS Number(65)', 'value': '64512'},
+                        {'capability': 'Route Refresh (Cisco)(128)', 'value': 'yes'}
+                    ]
+                },
+                'prefix-counter': {
+                    'entry': {
+                        '@afi-safi': 'bgpAfiIpv4-unicast',
+                        'incoming-total': '2',
+                        'incoming-accepted': '2',
+                        'incoming-rejected': '0',
+                        'policy-rejected': '0',
+                        'outgoing-total': '0',
+                        'outgoing-advertised': '0'
+                    }
+                }
+            }
+        }
+        ```
+
+        # Returns
+
+        dict: BGP peers information.
+
+        """
+
+        response = self.op_parser(
+            cmd="<show><routing><protocol><bgp><peer></peer></bgp></protocol></routing></show>", cmd_in_xml=True
+        )
```

Comment:
I would suggest we go with the same approach as in the other methods to keep the structure the same, passing the command instead of the actual XML API call, would be best to test for sure.
```suggestion
        response = self.op_parser(cmd="show routing protocol bgp peer")
```

Reply:
It was just my personal preference because in reality it's XML API that is queried, CLI command is just another "layer of abstraction". But if you think CLI command is better, I'm OK too. However, I think there were other methods using API call as well.